job_test.go

```diff
@@ -20,6 +22,23 @@ func TestTags(t *testing.T) {
 	assert.ElementsMatch(t, j.Tags(), []string{"tags", "tag", "some"})
 }
 
+func TestSingletonMode(t *testing.T) {
```

Comment:
this test itself is passing, and passes in a single test run, but when run in aggregate, the next test is failing. Seems the scheduler does in fact send out job that is then waiting on the singleflight to allow it to run, and that goroutine isn't properly stopped / canceled on `s.Stop()` - looking into this

Reply:
indeed. Adding 

```diff
	s.Stop()
+	time.Sleep(2 * time.Second)
```

sleep to the end of the test, confirms that a job is run after the scheduler is stopped.

Reply:
so, it appears because the scheduler is scheduling another job, and it's sent to the singleton which includes a wait group, even though the scheduler is stopped, the jobs aren't currently told about that stoppage.

I am interested in how best people view to solve this. I found *a* solution by adding a context with cancel into the job itself:

```golang
func TestSingletonMode(t *testing.T) {
	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()

	s := NewScheduler(time.UTC)
	var trigger int32
	job, _ := s.Every(1).Second().Do(func() {
		select {
		case <-ctx.Done():
			return
		default:
			if atomic.LoadInt32(&trigger) == 1 {
				t.Fatal("Restart should not occur")
			}
			atomic.AddInt32(&trigger, 1)
			fmt.Println("I am a long task")
			time.Sleep(3 * time.Second)
		}
	})
	job.SingletonMode()
	s.StartAsync()
	time.Sleep(2 * time.Second)
	cancel()
	s.Stop()
	time.Sleep(2 * time.Second)
}
```

If we go this route - we're essentially saying, you the caller are responsible for making sure your job doesn't run again if you stop the scheduler.

I don't know that there is a way to build it into the library because we can't force jobs to take in a context that we can cancel, or even that the job has to stop when it's cancelled.